client/dashboard/sites/settings-site-visibility/launch-form.tsx

```diff
@@ -1,12 +1,91 @@
+import { formatCurrency } from '@automattic/number-formatters';
+import { useQuery } from '@tanstack/react-query';
 import {
+	__experimentalHStack as HStack,
 	__experimentalText as Text,
 	__experimentalVStack as VStack,
 	Button,
+	ExternalLink,
 } from '@wordpress/components';
-import { __ } from '@wordpress/i18n';
+import { createInterpolateElement } from '@wordpress/element';
+import { __, _n, sprintf } from '@wordpress/i18n';
 import { addQueryArgs } from '@wordpress/url';
+import { agencyBlogQuery } from '../../app/queries';
 import type { Site } from '../../data/types';
 
+function useAgencyBillingMessage( site: Site ) {
+	const { data, isLoading: isLoading, isError: isError } = useQuery( agencyBlogQuery( site.ID ) );
+	if ( ! data ) {
+		return undefined;
+	}
+
+	const priceInfoIsDefined =
+		Number.isFinite( data.prices?.actual_price ) && typeof data.prices?.currency === 'string';
+
+	if ( isLoading || isError || ! priceInfoIsDefined ) {
+		return __( "After launch, we'll bill your agency in the next billing cycle." );
+	}
+
+	const { existing_wpcom_license_count: existingWPCOMLicenseCount = 0, name, prices } = data;
+	const price = formatCurrency( prices.actual_price, prices.currency );
+
+	return createInterpolateElement(
+		sprintf(
+			/* translators: agencyName is the name of the agency that will be billed for the site; licenseCount is the number of licenses the agency will be billed for; price is the price per license */
+			_n(
+				"After launch, we'll bill %(agencyName)s in the next billing cycle. With %(licenseCount)s production hosting license, you will be charged %(price)s / license / month. <learnMoreLink>Learn more.</learnMoreLink>",
```

Comment:
Not sure how much difference this makes, but seems like the format string should be

```suggestion
				"After launch, we'll bill %(agencyName)s in the next billing cycle. With %(licenseCount)d production hosting license, you will be charged %(price)s / license / month. <learnMoreLink>Learn more.</learnMoreLink>",
```

Reply:
Makes sense, updated in bd4f56984ca95b314e913c8ad7f01524e6cf2eb7.